alvr/server/src/connection.rs

```diff
@@ -642,10 +643,38 @@ fn try_connect(mut client_ips: HashMap<IpAddr, String>) -> ConResult {
         thread::spawn(|| ())
     };
 
-    let tracking_manager = Arc::new(Mutex::new(TrackingManager::new()));
+    let tracking_manager: Arc<
+        alvr_common::parking_lot::lock_api::Mutex<
+            alvr_common::parking_lot::RawMutex,
+            TrackingManager,
+        >,
+    > = Arc::new(Mutex::new(TrackingManager::new()));
+    let hand_gesture_manager: Arc<
+        alvr_common::parking_lot::lock_api::Mutex<
+            alvr_common::parking_lot::RawMutex,
+            HandGestureManager,
+        >,
+    > = Arc::new(Mutex::new(HandGestureManager::new()));
 
     let tracking_receive_thread = thread::spawn({
         let tracking_manager = Arc::clone(&tracking_manager);
+        let hand_gesture_manager = Arc::clone(&hand_gesture_manager);
+
+        let mut gestures_button_mapping_manager = if let Switch::Enabled(config) =
+            &SERVER_DATA_MANAGER.read().settings().headset.controllers
+        {
+            if let Switch::Enabled(_) = &config.hand_tracking.use_gestures {
+                Some(ButtonMappingManager::new_automatic(
+                    &HAND_GESTURE_PROFILE_INFO.button_set,
+                    &config.button_mapping_config,
+                ))
+            } else {
+                None
+            }
+        } else {
+            None
+        };
```

Comment:
I don't think having repeated `else None` is very good. You can collapse to a single if-let-else using a more complex matched pattern, or you can use `Switch::as_option()` and `.and_then()`  and `.map()` to contruct the final `Option` value.